File: TreasurersApp/Models/TransactionCategory.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace TreasurersApp.Models
{
    public partial class TransactionCategory
    {
        public TransactionCategory()
        {
            TransactionTypes = new HashSet<TransactionType>();
        }

        [JsonProperty("id")]
        public int TransactionCategoryId { get; set; }

        [JsonProperty("contributionCategoryName")]
        public string Name { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("displayOrder")]
        public int DisplayOrder { get; set; }

        [JsonProperty("active")]
        public bool? Active { get; set; }

        [JsonProperty("createdBy")]
        public System.Guid CreatedBy { get; set; } // CreatedBy

        [JsonProperty("createdDate")]
        public System.DateTime CreatedDate { get; set; } // CreatedDate

        [JsonProperty("lastModifiedBy")]
        public System.Guid LastModifiedBy { get; set; } // LastModifiedBy

        [JsonProperty("lastModifiedDate")]
        public System.DateTime LastModifiedDate { get; set; } // LastModifiedDate

        [JsonProperty("transactionTypes")]
        public ICollection<TransactionType> TransactionTypes { get; set; }
    }
}

File: TreasurersApp/ClientApp/src/app/models/TransactionCategory.ts
export class TransactionCategory {
  id: number = null;
  name: string = null;
  description: string = null;
  displayOrder: number = null;
  active = false;
  createdBy: string = null;
  createdDate: Date = null;
  lastModifiedBy: string = null;
  lastModifiedDate: Date = null;

  constructor(id: number,
    name: string,
    description: string,
    displayOrder: number,
    active: boolean,
    createdBy: string,
    createdDate: Date,
    lastModifiedBy: string,
    lastModifiedDate: Date) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.displayOrder = displayOrder;
    this.active = active;
    this.createdBy = createdBy;
    this.createdDate = createdDate;
    this.lastModifiedBy = lastModifiedBy;
    this.lastModifiedDate = lastModifiedDate;
  }
}
